Sources/Workspace/Workspace.swift

```diff
@@ -569,16 +569,18 @@ public class Workspace {
         reset: Bool = false
     ) throws {
         if reset {
-            try pinsStore.unpinAll()
+            pinsStore.unpinAll()
+            try pinsStore.saveState()
         }
         // Start pinning each dependency.
         for dependencyManifest in dependencyManifests.dependencies {
-            try pin(
-                pinsStore: pinsStore,
+            pin(pinsStore: pinsStore,
                 dependency: dependencyManifest.dependency,
                 package: dependencyManifest.manifest.name,
                 reason: reason)
         }
+        
+        try pinsStore.saveState()
```

Comment:
But `pinAll` is a public method of `Workspace`. Shouldn't it have the responsibility of saving the state the same way a method like `edit` does `managedDependencies.saveState()`?